### PR TITLE
Initialise project

### PR DESCRIPTION
**Pull request by robbiemcmichael**
The docs and tests could probably do with a bit of improvement, but the base functionality of the library is working as intended.

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,30 @@
+Copyright Author name here (c) 2019
```
Comment (carlosdagos) [inline LICENSE]: Probably put your name there 😄

File: src/Network/HTTP/Client/Overrides.hs [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+------------------------
+-- HTTP client overrides
+------------------------
+
+-- | If the @HTTP_CLIENT_OVERRIDES@ environment variable is set, this function
+-- reads the specified file as a 'ConfigFile' and applies the overrides to
+-- 'ManagerSettings'.
+withHttpClientOverrides :: HasCallStack => ManagerSettings -> IO ManagerSettings
+withHttpClientOverrides manager = do
+    filepath <- lookupEnv "HTTP_CLIENT_OVERRIDES"
```
Comment (carlosdagos) [inline src/Network/HTTP/Client/Overrides.hs:41]: Being able to set the path programmatically would be neat as well I think.

So you'd have three API functions in total

- `withHttpClientOverrides`: Reads the `HTTP_CLIENT_OVERRIDES` var.
- `withHttpClientOverridesFile`: Uses a `ConfigFile` as an argument.
- `withHttpClientOverridesFilePath`: Would accept a `FilePath` argument.

File: LICENSE (hunk 1 above)
Comment (robbiemcmichael) [reply LICENSE]: This is my legal name 😶

File: src/Network/HTTP/Client/Overrides.hs (hunk 2 above)
Comment (robbiemcmichael) [reply src/Network/HTTP/Client/Overrides.hs:41]: That's the idea behind combining `parseConfigFile` and `withHttpClientOverridesFile` which also allows the user to handle errors properly when parsing the config file. The function that uses an environment variable throws an exception if the config file fails to parse (I should make this clearer in the docs).

Comment (robbiemcmichael) [reply src/Network/HTTP/Client/Overrides.hs:41]: I've improved the docs and added `withHttpClientOverridesThrow` to make it clear that the function might throw an exception.

**Review by carlosdagos — APPROVED** [review]
Nice

**Pull request merged**